Source/Scene/GroundPolylinePrimitive.js

```diff
@@ -0,0 +1,856 @@
+                pickCommand.modelMatrix = modelMatrix;
+                pickCommand.boundingVolume = boundingSpheres[k];
+                pickCommand.cull = cull;
+
+                commandList.push(pickCommand);
+            }
+        }
+    }
+
+    /**
+     * Called when {@link Viewer} or {@link CesiumWidget} render the scene to
+     * get the draw commands needed to render this primitive.
+     * <p>
+     * Do not call this function directly.  This is documented just to
+     * list the exceptions that may be propagated when the scene is rendered:
+     * </p>
+     *
+     * @exception {DeveloperError} For synchronous GroundPolylinePrimitives, you must call GroundPolylinePrimitives.initializeTerrainHeights() and wait for the returned promise to resolve.
+     * @exception {DeveloperError} All instance geometries must have the same primitiveType.
+     * @exception {DeveloperError} Appearance and material have a uniform with the same name.
+     */
+    GroundPolylinePrimitive.prototype.update = function(frameState) {
+        if (!defined(this._primitive) && !defined(this.geometryInstances)) {
+            return;
+        }
+
+        if (!GroundPolylinePrimitive._initialized) {
+            //>>includeStart('debug', pragmas.debug);
+            if (!this.asynchronous) {
+                throw new DeveloperError('For synchronous GroundPolylinePrimitives, you must call GroundPolylinePrimitives.initializeTerrainHeights() and wait for the returned promise to resolve.');
+            }
+            //>>includeEnd('debug');
+
+            GroundPolylinePrimitive.initializeTerrainHeights();
+            return;
+        }
+
+        var i;
+
+        var that = this;
+        var primitiveOptions = this._primitiveOptions;
+        if (!defined(this._primitive)) {
+            var geometryInstances = isArray(this.geometryInstances) ? this.geometryInstances : [this.geometryInstances];
+            var geometryInstancesLength = geometryInstances.length;
+            var attributes;
+
+            // If using PolylineColorAppearance, check if each instance has a color attribute.
+            if (this.appearance instanceof PolylineColorAppearance) {
+                for (i = 0; i < geometryInstancesLength; ++i) {
+                    attributes = geometryInstances[i].attributes;
+                    if (!defined(attributes) || !defined(attributes.color)) {
+                        throw new DeveloperError('All GeometryInstances must have color attributes to use PolylineColorAppearance with GroundPolylinePrimitive.');
+                    }
+                }
+            }
+
+            // Automatically create line width attributes
+            for (i = 0; i < geometryInstancesLength; ++i) {
+                var geometryInstance = geometryInstances[i];
+                attributes = geometryInstance.attributes;
+                if (!defined(attributes.width)) {
+                    attributes.width = new GeometryInstanceAttribute({
```

Comment:
In general, we don't modify any objects that were parameters to a constructor or are set as properties. You'll need to clone the instances.